packages/datadog-shimmer/src/shimmer.js

```diff
@@ -33,28 +37,41 @@ function wrapFn (original, delegate) {
 
 function wrapMethod (target, name, wrapper) {
   assertMethod(target, name)
-  assertNotClass(target[name]) // TODO: support constructors of native classes
   assertFunction(wrapper)
 
   const original = target[name]
   const wrapped = wrapper(original)
   const descriptor = Object.getOwnPropertyDescriptor(target, name)
 
+  const attributes = {
+    configurable: true,
+    ...descriptor
+  }
+
+  copyProperties(original, wrapped)
+
   if (descriptor) {
     unwrappers.set(wrapped, () => Object.defineProperty(target, name, descriptor))
+
+    if (descriptor.get || descriptor.set) {
+      attributes.get = () => wrapped
+    } else {
+      attributes.value = wrapped
+    }
+
+    // TODO: create a single object for multiple wrapped methods
```

Comment:
Still needed?

Reply:
Yes, because write now multiple calls to `wrapMethod` are needed to patch multiple methods, and a new "proxy" object will be created each time.